applications/GeoMechanicsApplication/tests/cpp_tests/test_newmark_Pw_scheme.cpp

```diff
@@ -0,0 +1,141 @@
+{
+    NewmarkQuasistaticPwScheme<SparseSpaceType, LocalSpaceType> scheme(0.75);
+
+    Model model;
+    auto& model_part = model.CreateModelPart("dummy", 2);
+    model_part.AddNodalSolutionStepVariable(WATER_PRESSURE);
+    model_part.AddNodalSolutionStepVariable(DT_WATER_PRESSURE);
+    auto p_node = model_part.CreateNewNode(0, 0.0, 0.0, 0.0);
+
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN(scheme.Check(model_part),
+                                      "missing WATER_PRESSURE dof on node ")
+}
+
+KRATOS_TEST_CASE_IN_SUITE(ForMissingDtWaterPressureSolutionStepVariable_CheckNewmarkPwScheme_Throws,
+                          KratosGeoMechanicsFastSuite)
+{
+    NewmarkQuasistaticPwScheme<SparseSpaceType, LocalSpaceType> scheme(0.75);
+
+    Model model;
+    auto& model_part = model.CreateModelPart("dummy", 2);
+    model_part.AddNodalSolutionStepVariable(WATER_PRESSURE);
+    auto p_node = model_part.CreateNewNode(0, 0.0, 0.0, 0.0);
+    p_node->AddDof(WATER_PRESSURE);
+
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN(
+        scheme.Check(model_part),
+        "DT_WATER_PRESSURE variable is not allocated for node 0")
+}
+
+KRATOS_TEST_CASE_IN_SUITE(ForMissingWaterPressureSolutionStepVariable_CheckNewmarkPwScheme_Throws,
+                          KratosGeoMechanicsFastSuite)
+{
+    NewmarkQuasistaticPwScheme<SparseSpaceType, LocalSpaceType> scheme(0.75);
+
+    Model model;
+    auto& model_part = model.CreateModelPart("dummy", 2);
+    model_part.AddNodalSolutionStepVariable(DT_WATER_PRESSURE);
+    auto p_node = model_part.CreateNewNode(0, 0.0, 0.0, 0.0);
+
+    KRATOS_EXPECT_EXCEPTION_IS_THROWN(
+        scheme.Check(model_part),
+        "WATER_PRESSURE variable is not allocated for node 0")
+}
+
+KRATOS_TEST_CASE_IN_SUITE(NewmarkPwSchemeUpdate_SetsDtPressure, KratosGeoMechanicsFastSuite)
+{
+    auto scheme = CreateValidScheme();
+    Model model;
+    ModelPart& model_part = CreateValidModelPart(model);
+
+    constexpr double current_pressure = 10.0;
+    constexpr double previous_pressure = 5.0;
+    constexpr double previous_dt_pressure = 3.0;
+    constexpr double delta_time = 2.0;
+
+    model_part.GetProcessInfo()[DELTA_TIME] = delta_time;
+    Node& node = model_part.Nodes()[0];
+    node.FastGetSolutionStepValue(WATER_PRESSURE, 0) = current_pressure;
+    node.FastGetSolutionStepValue(WATER_PRESSURE, 1) = previous_pressure;
+    node.FastGetSolutionStepValue(DT_WATER_PRESSURE, 1) = previous_dt_pressure;
+
+    ModelPart::DofsArrayType dof_set;
+    CompressedMatrix A;
+    Vector Dx;
+    Vector b;
+
+    scheme.Initialize(model_part);
+    scheme.Predict(model_part, dof_set, A, Dx, b);
+
+    // This is the expected value as calculated by the UpdateVariablesDerivatives
+    KRATOS_EXPECT_DOUBLE_EQ(node.FastGetSolutionStepValue(DT_WATER_PRESSURE, 0), 7.0 / 3.0);
+}
+
+KRATOS_TEST_CASE_IN_SUITE(InitializeNewmarkPwScheme_SetsTimeFactors, KratosGeoMechanicsFastSuite)
+{
+    auto scheme = CreateValidScheme();
```

Comment:
When you inline this function, you can move `constexpr double theta = 0.75;` to the top of the test body and use it when creating the scheme. That will make it more obvious that the numbers are correct.

Reply:
Done